### PR TITLE
Update tidy check for error codes testing

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Now that all the error codes have been grouped into one file, we moved the long error explanations into markdown files, which means that we need to read those markdown file to also count their code blocks as well.

So 2 "big" things happened here:
* No more need to keep the current error code when reading the `error_codes.rs` file since it's been put into the equivalent markdown file.
* Need to instead read the markdown file (but it's simpler since I can just look for code blocks directly).

### CODE REVIEW

File: src/tools/tidy/src/error_codes_check.rs [hunk 1]
```diff
@@ -50,41 +52,75 @@ const WHITELIST: &[&str] = &[
...
+fn check_error_code_explanation(
+    f: &str,
+    error_codes: &mut HashMap<String, bool>,
+    err_code: String,
+) {
+    for line in f.lines() {
+        let s = line.trim();
+        if s.starts_with("```") && s.contains("compile_fail") && s.contains('E') {
+            let nb = error_codes.entry(err_code).or_insert(false);
+            *nb = true;
```
Comment (Mark-Simulacrum) [inline src/tools/tidy/src/error_codes_check.rs]: What is the point of this `or_insert(false)` and then always setting it to true? Why not just `insert(err_code, true)`?

Comment (GuillaumeGomez) [reply src/tools/tidy/src/error_codes_check.rs]: Good point! Updating.

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by Mark-Simulacrum** [comment]
Could you also say a bit about what this is doing? i.e., what is the goal of this PR, what are we trying to change, and why? I don't have the time to dig into that myself so it'd make an r+ easier on my end.

**Comment by GuillaumeGomez** [comment]
Sure. Now that all the error codes have been grouped into one file, we moved the long error explanations into markdown files, which means that we need to read those markdown file to also count their code blocks as well.

So 2 "big" things happened here:
* No more need to keep the current error code when reading the `error_codes.rs` file since it's been put into the equivalent markdown file.
* Need to instead read the markdown file (but it's simpler since I can just look for code blocks directly).

File: src/tools/tidy/src/error_codes_check.rs [hunk 2]
```diff
@@ -50,41 +52,74 @@ const WHITELIST: &[&str] = &[
...
-        } else if s.starts_with("```") && s.contains("compile_fail") && s.contains('E') {
-            if let Some(err_code) = s.splitn(2, 'E').skip(1).next() {
-                if let Some(err_code) = err_code.splitn(2, ',').next() {
-                    let nb = error_codes.entry(format!("E{}", err_code)).or_insert(false);
-                    *nb = true;
+                // Now we extract the tests from the markdown file!
+                let md = some_or_continue!(s.splitn(2, "include_str!(\"").skip(1).next());
+                let md_file_name = some_or_continue!(md.splitn(2, "\")").next());
+                let path = some_or_continue!(path.parent()).join(md_file_name);
+                match File::open(&path) {
```
Comment (Mark-Simulacrum) [inline src/tools/tidy/src/error_codes_check.rs]: Could we use `fs::read_to_string` here? Seems like there's no reason why not?

**Comment by Mark-Simulacrum** [comment]
For the future it'd be great to include that information in the commit; I've copied it into the PR description.

r=me with last nit fixed

**Comment by GuillaumeGomez** [comment]
Indeed, sorry...

**Pull request merged**